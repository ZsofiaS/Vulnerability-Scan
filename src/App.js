import React from 'react';
import './App.css';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <p>
          Vulnerability Scan
        </p>
      </header>
    </div>
  );
}

export default App;
