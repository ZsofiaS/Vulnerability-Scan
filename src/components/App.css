* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}
h2 {
  text-align: left;
}
.App {
  text-align: left;
  margin: 0;
  padding: 0;
  background-color: white;
  min-height: 100vh;
}
.topContainer {
  display: flex;
}


/* .App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
} */
