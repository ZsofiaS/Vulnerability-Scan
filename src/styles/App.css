/* GENERAL */
* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}
h1 {
  margin: 2rem auto;
  font-size: 1.8rem;
}
h2 {
  text-align: left;
  margin: 1rem 0;
  font-size: 1.5rem;
}
.App {
  text-align: left;
  margin: 0;
  padding: 0;
  background-color: white;
  min-height: 100vh;
}

/* TOP BAR */
.topBar {
  border-bottom: 1px solid gainsboro;
  padding: 0.8rem 2rem;
  background-color: #4D5656;
  color: white;
}
.rightMenu {
  float: right;
}
    /* ACCOUNT */
.accountContainer {
  position: relative;
  display: inline-block;
}
.accountDropdown {
  font-size: 0.85rem;
  background-color: white;
  display: block;
  position: absolute;
  right: 0;
  top: 2.5rem;
  z-index: 1;
  width: 25rem;
  border: 1px solid gainsboro;
  padding: 0.3rem 0.5rem;
  color: black;
}
.account {
  margin: 0.5rem 0;
}
    /* NOTIFICATIONS */
.notificationsContainer {
  position: relative;
  display: inline-block;
}
.notificationsDropdown {
  font-size: 0.85rem;
  background-color: white;
  display: block;
  position: absolute;
  right: -1rem;
  top: 2.5rem;
  z-index: 1;
  width: 25rem;
  border: 1px solid gainsboro;
  padding: 0.3rem 0.5rem;
  color: black;
}
.bell {
  margin-right: 1rem;
}
.read-false {
  font-weight: bold;
}
.message {
  display: flex;
  justify-content: space-between;
  margin: 0.5rem 0;
}

/* SUMMARY */
.summary {
  margin: 0 8rem;
  padding: 1rem 2rem;
  text-align: center;
}
.summaryContainer {
  display: flex;
  justify-content: space-around;
  text-align: left;
}
.status, .assets, .severities {
  border: 1px solid #F2F3F4;
  padding: 1rem;
  background-color: #F4F6F6;
  flex-basis: 100%;
}
.assets {
  margin: 0 1rem;
}
.assets ul, .severities ul {
  padding: 0;
}
.asset {
  list-style-type: none;
}
    /* SEVERITY ITEM */
.severityContainer {
  display: flex;
  justify-content: space-between;
}
.severity {
  margin-left: 0.5rem;
  padding: 0.1rem 0.3rem;
  display: flex;
  justify-content: flex-end;
}
.zero {
  background-color: none;
  margin-left: 1rem;
  margin-right: 0.8rem;
}
li {
  list-style-type: none;
  display: flex;
  justify-content: flex-start;
  margin-top: 0.5rem;
  padding-left: 1rem;
}

/* DETAILS */
.details {
  margin: 1rem 8rem 0 8rem;
  padding: 1rem 2rem;
  text-align: center;
}
    /* TABLE */
table {
  table-layout: fixed;
  width: 100%;
  border-collapse: collapse;
  text-align: center;
}
thead {
  background-color: #F4F6F6;
}
th, td {
  padding: 0.5rem 0.3rem;
}
    /* TABLEROW */
.mainRow {
  border-top: 1px solid gainsboro;
}
.secondaryRow {
  text-align: justify;
}
.infoButton {
  border: none;
  background-color: white;
}
.infoButton:hover {
  cursor: pointer;
}
.warning {
  padding: 0.3rem;
}
.critical {
  background-color: fireBrick;
}
.high {
  background-color: orangeRed;
}
.medium {
  background-color: orange;
}
.low {
  background-color: yellow;
}
.information {
  background-color: #3498DB;
}
