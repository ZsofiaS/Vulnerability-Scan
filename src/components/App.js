import React from 'react';
import './App.css';
import Summary from './Summary';
import Details from './Details';
import User from './User';

const App = () => {
  return (
    <div className="App">
      <header className="header">
        <User />
      </header>
      <div className="topContainer">
        <Summary />
      </div>
      <Details />
    </div>
  );
}

export default App;
