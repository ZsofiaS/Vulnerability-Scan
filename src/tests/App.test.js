import React from 'react';
import { shallow } from 'enzyme';
import App from '../components/App';
import toJson from 'enzyme-to-json';

describe('renders App component', () => {
  it('renders without crashing', () => {
    shallow(<App />);
  });
});

describe('snapshot', () => {
  it('App snapshot', () => {
    const tree = shallow(<App />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
