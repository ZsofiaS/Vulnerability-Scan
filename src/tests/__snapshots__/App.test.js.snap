// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot App snapshot 1`] = `
<div
  className="App"
>
  <TopBar />
  <Summary />
  <Details />
</div>
`;
